mdbranch

AfroditiMariaZaki opened the pull request:
Added all the files required for the MD tutorial.

Fixes #37

AfroditiMariaZaki committed:
Create CONTRIBUTORS.md

AfroditiMariaZaki committed:
Update README.md

AfroditiMariaZaki committed:
Update README.md

AfroditiMariaZaki committed:
Add files via upload

Lecture slides for Molecular Dynamics (.pdf)

AfroditiMariaZaki committed:
Update README.md

AfroditiMariaZaki committed:
Update README.md

AfroditiMariaZaki committed:
Create README.md

AfroditiMariaZaki committed:
Create README.md

AfroditiMariaZaki committed:
Create README.md

AfroditiMariaZaki committed:
Create README.md

AfroditiMariaZaki committed:
Create README.md

AfroditiMariaZaki committed:
Add files via upload

AfroditiMariaZaki committed:
Create README.md

AfroditiMariaZaki committed:
Add files via upload

AfroditiMariaZaki committed:
Update README.md

AfroditiMariaZaki committed:
Create README.md

AfroditiMariaZaki committed:
Add files via upload

AfroditiMariaZaki committed:
Create README.md

AfroditiMariaZaki committed:
Add files via upload

AfroditiMariaZaki committed:
Add files via upload

AfroditiMariaZaki committed:
Add files via upload

AfroditiMariaZaki committed:
Update README.md

IAlibay commented on README.md:
```diff
@@ -7,6 +7,12 @@ This two and half day course is aimed at graduate students who have had no prior
+
+- Setup, equilibrate and simulate a protein-ligand system in water.
+- Visualise the system and its trajectory with [VMD](https://www.ks.uiuc.edu/Research/vmd/).
+- Perform basic analysis utilising implemented GROMACS tools and the [xmgrace](http://plasma-gate.weizmann.ac.il/Grace/doc/UsersGuide.html) plotting tool.
```
`utilising implemented GROMACS tools` would this be better as `utilising GROMACS tools`? (I'm mainly querying the necessity of `implemented` in the sentence.

IAlibay commented on tutorials/MD/CONTRIBUTORS.md:
```diff
@@ -0,0 +1,9 @@
+# List of contributors to the Oxford Computational Biochemitry molecular dynamics course
```
Capitalise Molecular Dynamics?

IAlibay commented on tutorials/MD/README.md:
```diff
@@ -1 +1,28 @@
+
+## Contents
+
+The Oxford Computational Biochemistry Molecular Dynamics tutorial utilises the  GROMACS simulation package and describes the steps for setting up, equilibrating and simulating a protein-ligand system in water. 
```
double space between `the` and `GROMACS`? (could just be github).

IAlibay commented on tutorials/MD/README.md:
```diff
@@ -1 +1,28 @@
+
+The tutorial also makes use of the VMD software for visualisation purposes, Gromacs tools for performing basic analysis and the xmgrace package for plotting.
+
+The md-prac.html file contains the detailed instructions required for the tutorial (and have also been handed out) and can be opened with a browser by typing on a terminal:
```
`(and have also been handed out)` since utlimately we want this tutorial to be available to everyone maybe it's worth precising that we mean during the live tutorial, so maybe `(which have also been handed out to you if you are attending the tutorial in person)` or something along those lines?

IAlibay commented on tutorials/MD/README.md:
```diff
@@ -1 +1,28 @@
+
+The md-prac.html file contains the detailed instructions required for the tutorial (and have also been handed out) and can be opened with a browser by typing on a terminal:
+
+firefox md-prac.html
```
Turn to code block here?
Something like:
 `firefox md-prac.html`
To highlight that it is a command?

IAlibay commented on tutorials/MD/README.md:
```diff
@@ -1 +1,28 @@
+
+firefox md-prac.html
+
+The lecture slides are also included in .pdf format.
```
To aid comprehension, would it be worth giving the name of the pdf file?

IAlibay commented on tutorials/MD/README.md:
```diff
@@ -1 +1,28 @@
+The lecture slides are also included in .pdf format.
+
+## Requirements
+The following packages were used: 
```
packages -> software packages?
(pedantic change but mainly thinking it might be easier to comprehend to a new starter)?

IAlibay commented on tutorials/MD/README.md:
```diff
@@ -1 +1,28 @@
+## Requirements
+The following packages were used: 
+
+Gromacs version 2018.
```
link to the gromacs website?

IAlibay commented on tutorials/MD/README.md:
```diff
@@ -1 +1,28 @@
+The following packages were used: 
+
+Gromacs version 2018.
+VMD version 1.9.2.
```
link to the vmd website?

IAlibay commented on tutorials/MD/README.md:
```diff
@@ -1 +1,28 @@
+
+Gromacs version 2018.
+VMD version 1.9.2.
+Grace version 5.1.25
```
link to the grace website?

IAlibay commented on tutorials/MD/data/em.mdp:
```diff
@@ -0,0 +1,314 @@
+; e.g.: -I/home/joe/doe -I/home/mary/roe
+include                  = 
+; e.g.: -DPOSRES -DFLEXIBLE (note these variable names are case sensitive)
+define                   = 
```
Is it a good idea to have empty variable assignements here? As far as I am aware these mean nothing and (need to check) I think 2020 might even raise them as warnings now.

IAlibay commented on tutorials/MD/data/em.mdp:
```diff
@@ -0,0 +1,314 @@
+; Scaling of reference coordinates, No, All or COM
+refcoord-scaling         = No
+
+; OPTIONS FOR QMMM calculations
```
I would suggest removing unecessary defaults, e.g the QM/MM keywords, in my opinion it makes mdp comprehension harder?

IAlibay commented on tutorials/MD/data/md.mdp:
```diff
@@ -0,0 +1,314 @@
+; Scaling of reference coordinates, No, All or COM
+refcoord_scaling         = No
+
+; OPTIONS FOR QMMM calculations
```
As per my em.mdp comments on this section.

IAlibay commented on tutorials/MD/prerun/README.md:
```diff
@@ -0,0 +1 @@
+You can find all the data for the simulation that has already run.
```
Worth adding a title line?

IAlibay commented on tutorials/MD/data/md.mdp:
```diff
@@ -0,0 +1,314 @@
+gen-seed                 = 173529
+
+; OPTIONS FOR BONDS    
+constraints              = all-bonds
```
So maybe worth a longer discussion here, but it would definitely be worth highlighting in teaching these mdps the issues with the historical use of `all-bonds` constraints and why they are not recommended for most modern ffs (in this case I think gromos might have actually been paramterised for it, someone else can check here). In most cases `all-bond` constraints only serve to add an extra layer of atrificiality to the system without any performance gains (it is my understanding that `all-bond` constraints actually run slower than `h-bonds`). That being said there are cases where `all-bonds` are necessary, primarily when using VS or HMR methods to get dt to ~ 4 fs.

I'll refer back to this change in grompp behaviour as of gromacs 2019: http://manual.gromacs.org/documentation/2019/release-notes/2019/major/miscellaneous.html#grompp-discourages-use-of-constraints-all-bonds

Probably too late to address for monday, but tagging @philbiggin here for future iterations of the course.

IAlibay commented on tutorials/MD/prerun/setup/mdout.mdp:
```diff
@@ -0,0 +1,353 @@
+; Scaling of reference coordinates, No, All or COM
+refcoord-scaling         = No
+
+; OPTIONS FOR QMMM calculations
```
As above in em.mdp.

IAlibay commented on tutorials/MD/run/README.md:
```diff
@@ -0,0 +1 @@
+Perform the MD production run in this directory.
```
Title?

IAlibay commented on tutorials/MD/setup/README.md:
```diff
@@ -0,0 +1 @@
+Setup the protein system and energy minimise it in this directory.
```
Title?

IAlibay reviewed (changes requested):
Looks good.

A sentence restructure here and there and I might be pendantic but I quite like for README.md files to have titles even if they only contain a single line.

Otherwise the only main thing is a general comment on my preference on maybe not including empty/unused default variable declarations unless necessary (makes mdps harder to read).

AfroditiMariaZaki committed:
Update README.md

AfroditiMariaZaki committed:
Update CONTRIBUTORS.md

AfroditiMariaZaki committed:
Update README.md

AfroditiMariaZaki committed:
Update README.md

AfroditiMariaZaki committed:
Update README.md

AfroditiMariaZaki committed:
Update README.md

AfroditiMariaZaki committed:
Update README.md

AfroditiMariaZaki committed:
Update README.md

AfroditiMariaZaki committed:
Update README.md

AfroditiMariaZaki committed:
Update README.md

AfroditiMariaZaki committed:
Update README.md

AfroditiMariaZaki committed:
Update README.md

AfroditiMariaZaki committed:
Update README.md

AfroditiMariaZaki committed:
Update README.md

IAlibay commented:
@AphroditiMaria let me know when you want me to merge

AfroditiMariaZaki committed:
Delete md-prac.html

AfroditiMariaZaki committed:
Add files via upload

AfroditiMariaZaki committed:
Delete md-prac.pdf

AfroditiMariaZaki committed:
Add files via upload

AfroditiMariaZaki commented:
I implemented the changes except for the ones regarding the mdp files, as I did not want to risk modifying something by mistake that may mess up the tutorial. I also deleted and re-uploaded the md-prac.html and the md-prac.pdf files, so I am not sure if you can see these changes or I need a new pull request for that? If not, you can merge now if you want, thanks!

IAlibay commented:
I'll merge as-is, we should consider the .mdp option changes for next year's tutorial.
